src/components/Editor/tests/DebugLine.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { shallow } from "enzyme";
+import DebugLine from "../DebugLine";
+jest.mock(
+  "../../../utils/editor/source-documents",
+  jest.fn(() => ({
+    getDocument: jest.fn()
+  }))
+);
+import { getDocument } from "../../../utils/editor/source-documents";
+getDocument.mockImplementation(() => ({
+  addLineClass: jest.fn(),
+  removeLineClass: jest.fn()
+}));
+
+const DebugLineComponent = React.createFactory(DebugLine);
+
+function generateDefaults(overrides) {
+  return {
+    editor: {
+      codeMirror: {
+        markText: () => ({ clear: jest.fn() })
+      }
+    },
+    selectedFrame: {
+      location: {
+        sourceId: "x",
+        line: 2
+      }
+    },
+    ...overrides
+  };
+}
+
+function render(overrides = {}) {
+  const props = generateDefaults(overrides);
+  const component = shallow(new DebugLineComponent(props));
+  return { component, props };
+}
+
+describe("DebugLine Component", () => {
+  describe("mount", () => {
+    it("should keep the debugExpression state", async () => {
+      const { component } = render();
+      expect(component.state().debugExpression).toBeDefined();
+      expect(component.state().debugExpression.clear).toBeDefined();
+    });
+  });
+
+  describe("unmount", () => {
+    it("should remove the debug line", async () => {
+      const { component, props: { editor } } = render();
+      component.unmount();
+      expect(editor.codeMirror.removeLineClass).toHaveBeenCalled();
+    });
+
+    it("should clear the debug line", async () => {
+      const { component, props: { editor } } = render();
+      component.unmount();
+      expect(editor.codeMirror.removeLineClass).toHaveBeenCalled();
+    });
+  });
+
+  describe("update", () => {
+    const selectedLocation = {
+      location: {
+        sourceId: "x",
+        line: 1
+      }
+    };
+
+    fit("should remove the old debug line", async () => {
+      const { component } = render();
+      component.setProps({ selectedLocation });
+      const getDoc = getDocument();
```

Comment:
do we want to assert we get the right document... it might be nice to simulate setting a new document on codemirror and make sure we get the document for the correct source